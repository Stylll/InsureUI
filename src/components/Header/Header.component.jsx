import React, { useState } from 'react';
import Modal from '../Modal/Modal.component';
import CreateItem from '../CreateItem/CreateItem.component';
import './Header.styles.scss';

const Header = () => {
  const [showCreateItemModal, setShowCreateItemModal] = useState(false);

  const showCreateItem = () => {
    setShowCreateItemModal(true);
  };

  const hideCreateItem = () => {
    setShowCreateItemModal(false);
  };

  return (
    <div>
      <div className="header container">
        <div className="left">
          <div className="link">
            Insure App
          </div>
        </div>
        <div className="right">
          <div className="link" onClick={showCreateItem} data-testid="add-item">
            Add Item
          </div>
        </div>
      </div>
      <Modal isModalVisible={showCreateItemModal}
        handleModalClose={hideCreateItem}
      >
        <CreateItem closeModal={hideCreateItem} />
      </Modal>
    </div>
  );
};

export default Header;
