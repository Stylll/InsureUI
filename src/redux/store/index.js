import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import createSagaMiddleware from 'redux-saga';

import rootReducer from '../reducers';
import rootSaga from '../sagas/index';

const sagaMiddleware = createSagaMiddleware();
const middleware = composeWithDevTools(applyMiddleware(sagaMiddleware));
const store = createStore(rootReducer, middleware);

sagaMiddleware.run(rootSaga);

export default store;
