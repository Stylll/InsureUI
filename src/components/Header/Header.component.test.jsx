import { render, fireEvent } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import Header from './Header.component';
import store from '../../redux/store';

describe('Header Test', () => {
  it('should render properly', () => {
    const { baseElement } = render(
      <Provider store={store}>
        <Header />
      </Provider>,
    );
    expect(baseElement).toMatchSnapshot();
  });

  it('should render and close create item form', () => {
    const { getByTestId, getByText, queryByText } = render(
      <Provider store={store}>
        <Header />
      </Provider>,
    );
    fireEvent.click(getByTestId('add-item'));
    expect(getByText('Add an Item')).toBeInTheDocument();
    fireEvent.click(getByTestId('modal-close'));
    expect(queryByText('Add an Item')).not.toBeInTheDocument();
  });
});
