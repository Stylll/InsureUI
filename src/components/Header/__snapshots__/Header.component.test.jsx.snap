// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Test should render properly 1`] = `
<body>
  <div>
    <div>
      <div
        class="header container"
      >
        <div
          class="left"
        >
          <div
            class="link"
          >
            Insure App
          </div>
        </div>
        <div
          class="right"
        >
          <div
            class="link"
            data-testid="add-item"
          >
            Add Item
          </div>
        </div>
      </div>
      <div
        class="modal"
      />
    </div>
  </div>
</body>
`;
